kong/resty/dns/client.lua

```diff
@@ -651,7 +651,9 @@ _M.init = function(options)
 
   config = options -- store it in our module level global
 
-  resolve_max_wait = options.timeout / 1000 * options.retrans -- maximum time to wait for the dns resolver to hit its timeouts
+  -- maximum time to wait for the dns resolver to hit its timeouts
+  -- + 1s to ensure some delay in timer execution and semaphore return are accounted for
+  resolve_max_wait = options.timeout / 1000 * options.retrans + 1
```

Comment:
What could happen if we didn't include this `+1` ?

Reply:
Please see the comment above, without the +1 there is a risk semaphore might timeout before resolve actually happened.

Reply:
There are two scenerio where waiting time or timeout
 1. `options.timeout`: a timeout used for network IO of the DNS query
 2. `resolve_max_wait`: waiting for the completion of the DNS query.

If no +1s, these two timeouts will be similar. Consequently, the second scenario may conclude before the first DNS query retrieves the result.